sklearn/utils/_weight_vector.pyx.tp

```diff
@@ -30,57 +52,69 @@ cdef class WeightVector(object):
 
     Attributes
     ----------
-    w : ndarray, dtype=double, order='C'
+    w : ndarray, dtype={{c_type}}, order='C'
         The numpy array which backs the weight vector.
-    aw : ndarray, dtype=double, order='C'
+    aw : ndarray, dtype={{c_type}}, order='C'
         The numpy array which backs the average_weight vector.
-    wscale : double
+    w_data_ptr : {{c_type}}*
+        A pointer to the data of the numpy array.
+    wscale : {{c_type}}
         The scale of the vector.
     n_features : int
         The number of features (= dimensionality of ``w``).
-    sq_norm : double
+    sq_norm : {{c_type}}
         The squared norm of ``w``.
     """
-    def __cinit__(self, double [::1] w, double [::1] aw):
+
+    cdef readonly np.ndarray w
+    cdef readonly np.ndarray aw
+
+    def __cinit__(self,
+                  np.ndarray[{{c_type}}, ndim=1, mode='c'] w,
+                  np.ndarray[{{c_type}}, ndim=1, mode='c'] aw):
```

Comment:
is it necessary to replace the memory views ? Can't you do ``{{c_type}}[::1] w`` ?